src/audio_core/cubeb_input.cpp

```diff
@@ -56,6 +57,23 @@ void CubebInput::StartSampling(const Frontend::Mic::Parameters& params) {
     is_sampling = true;
 
     cubeb_devid input_device = nullptr;
+    if (device_id != Frontend::Mic::default_device_name && !device_id.empty()) {
+        cubeb_device_collection collection;
+        if (cubeb_enumerate_devices(impl->ctx, CUBEB_DEVICE_TYPE_INPUT, &collection) != CUBEB_OK) {
+            LOG_WARNING(Audio, "Audio input device enumeration not supported");
+        } else {
+            const auto collection_end = collection.device + collection.count;
+            const auto device =
+                std::find_if(collection.device, collection_end, [&](const cubeb_device_info& info) {
```

Comment:
This can be narrowed in scope down to only capturing `this`.

```suggestion
                std::find_if(collection.device, collection_end, [this](const cubeb_device_info& info) {
```